File: programmers/문자열 내 마음대로 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
bool compared(pair<string,string> a,pair<string,string> b){
    if(a.second==b.second)return a.first<b.first;
    return a.second<b.second;
}
vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;
    vector<pair<string,string>>hash;
    for(int i=0;i<strings.size();++i){
        hash.push_back(make_pair(strings[i],strings[i].substr(n,1)));
    }
    sort(hash.begin(),hash.end(),compared);
    for(int i=0;i<hash.size();++i){
        answer.push_back(hash[i].first);
    }
    return answer;
}

File: programmers/Level 1/문자열 내 마음대로 정렬하기.js
function solution(strings, n) {
    return strings.sort((a,b)=>{
        let first = a.charCodeAt(n);
        let second = b.charCodeAt(n);
        
        if(first === second)
            return a.localeCompare(b);
        else return first -second;
    })
}
